### PR TITLE
[GraphBolt] Add error messages for attribute size check

### PR DESCRIPTION
**Pull request by Skeleton003**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: graphbolt/src/fused_csc_sampling_graph.cc [hunk 1]
```diff
@@ -97,12 +97,21 @@ c10::intrusive_ptr<FusedCSCSamplingGraph> FusedCSCSamplingGraph::Create(
   }
   if (node_attributes.has_value()) {
     for (const auto& pair : node_attributes.value()) {
-      TORCH_CHECK(pair.value().size(0) == indptr.size(0) - 1);
+      TORCH_CHECK(
+          pair.value().size(0) == indptr.size(0) - 1,
+          "Expected node_attribute.size(0) and num_nodes to be euqal, "
```
Comment (mfbalin) [inline graphbolt/src/fused_csc_sampling_graph.cc:102]:
```suggestion
          "Expected node_attribute.size(0) and num_nodes to be equal, "
```

**Comment by mfbalin** [comment]
I didn't know TORCH_CHECK supports such mixed parameters for the error message. Really convenient.

File: graphbolt/src/fused_csc_sampling_graph.cc [hunk 2]
```diff
@@ -97,12 +97,21 @@ c10::intrusive_ptr<FusedCSCSamplingGraph> FusedCSCSamplingGraph::Create(
   }
   if (node_attributes.has_value()) {
     for (const auto& pair : node_attributes.value()) {
-      TORCH_CHECK(pair.value().size(0) == indptr.size(0) - 1);
+      TORCH_CHECK(
+          pair.value().size(0) == indptr.size(0) - 1,
```
Comment (mfbalin) [inline graphbolt/src/fused_csc_sampling_graph.cc:101]:
```suggestion
          pair.value().size(0) == NumNodes(),
```

File: graphbolt/src/fused_csc_sampling_graph.cc [hunk 3]
```diff
@@ -97,12 +97,21 @@ c10::intrusive_ptr<FusedCSCSamplingGraph> FusedCSCSamplingGraph::Create(
...
+          "Expected node_attribute.size(0) and num_nodes to be euqal, "
+          "but node_attribute.size(0) was ",
+          pair.value().size(0), ", and num_nodes was ", indptr.size(0) - 1,
```
Comment (mfbalin) [inline graphbolt/src/fused_csc_sampling_graph.cc:104]:
```suggestion
          pair.value().size(0), ", and num_nodes was ", NumNodes(),
```

File: graphbolt/src/fused_csc_sampling_graph.cc [hunk 4]
```diff
@@ -97,12 +97,21 @@ c10::intrusive_ptr<FusedCSCSamplingGraph> FusedCSCSamplingGraph::Create(
...
+          ".");
     }
   }
   if (edge_attributes.has_value()) {
     for (const auto& pair : edge_attributes.value()) {
-      TORCH_CHECK(pair.value().size(0) == indices.size(0));
+      TORCH_CHECK(
+          pair.value().size(0) == indices.size(0),
```
Comment (mfbalin) [inline graphbolt/src/fused_csc_sampling_graph.cc:111]:
```suggestion
          pair.value().size(0) == NumEdges(),
```

File: graphbolt/src/fused_csc_sampling_graph.cc [hunk 5]
```diff
@@ -97,12 +97,21 @@ c10::intrusive_ptr<FusedCSCSamplingGraph> FusedCSCSamplingGraph::Create(
...
+          "Expected edge_attribute.size(0) and num_edges to be equal, "
+          "but edge_attribute.size(0) was ",
+          pair.value().size(0), ", and num_edges was ", indices.size(0), ".");
```
Comment (mfbalin) [inline graphbolt/src/fused_csc_sampling_graph.cc:114]:
```suggestion
          pair.value().size(0), ", and num_edges was ", NumEdges(), ".");
```

**Review by mfbalin — COMMENTED** [review]
Making some suggestions to make use of the existing NumNodes and NumEdges functions, and fix a small typo. Otherwise, LGTM.

File: graphbolt/src/fused_csc_sampling_graph.cc (hunk 5 above)
Comment (Skeleton003) [reply graphbolt/src/fused_csc_sampling_graph.cc:114]: It is indeed convenient to make use of existing APIs, but the purpose of this function is to check all arguments necessary to create a FusedCSCSamplingGraph, which means we need to check not only the size of node/edge attributes, but also the size of `indptr` and `indices`. Therefore I insist not to use `NumNodes()` and `NumEdges()` here.

Comment (mfbalin) [reply graphbolt/src/fused_csc_sampling_graph.cc:114]: https://github.com/dmlc/dgl/blob/af0b63eda6315dcd0db6bdb25f2db84a0db12754/graphbolt/include/graphbolt/fused_csc_sampling_graph.h#L109-L110
https://github.com/dmlc/dgl/blob/af0b63eda6315dcd0db6bdb25f2db84a0db12754/graphbolt/include/graphbolt/fused_csc_sampling_graph.h#L112-L113

Comment (mfbalin) [reply graphbolt/src/fused_csc_sampling_graph.cc:114]: Either way is fine by me, it is a small nit.

**Review by mfbalin — APPROVED** [review]
LGTM with a small nit comment.

**Pull request merged**